app/javascript/components/editor/TestSummary.tsx

```diff
@@ -60,13 +60,13 @@ export function TestSummary({
         {isPresent(test.message) ? (
           <div className="--info">
             <h3>{messageLabels[test.status]}</h3>
-            <pre dangerouslySetInnerHTML={{ __html: test.messageHtml }}></pre>
+            <pre dangerouslySetInnerHTML={{ __html: test.messageHtml }} />
```

Comment:
This still seems problematic... what is the content of `test.messageHtml` exactly in this case?  Is there a non-zero chance that it can contain ANY user generated content at all (custom test errors, etc)?  Is it guaranteed to be pre-escaped by the test runner itself?

Reply:
So far in my experience `dangerouslySetInnerHTML` should require like the sign-off of 5 different people before it's approved. :-)

Reply:
Allowing test runners to generate HTML at all seems problematic to me from a security standpoint. _Test runners should generate JSON data that the website can interpret in a secure fashion, not raw HTML._

Reply:
They do.

Reply:
However, they are preprocessed before being displayed (e.g. they ansi is converted to colours)

Reply:
That's fine. But RAW html from test runners should probably not be allowed/escaped.